### PR TITLE
Update docs on Record and unknown types

### PR DESCRIPTION
**Pull request by markcowl**

### CODE REVIEW

**Review by timotheeguerin — APPROVED** [review]

File: docs/language-basics/built-in-types.md [hunk 1]
```diff
@@ -49,8 +49,8 @@ Built in types are related to each other according to the rules described in [ty
 | `boolean`   | Boolean with `true` and `false` values                                                     |
 | `null`      | Null value                                                                                 |
 | `Array<T>`  | Array model type, equivalent to `T[]`                                                      |
-| `Record<T>` | Model with string properties where all the properties have type `T`                        |
-| `unknown`   | A top type in TypeSpec that all types can be assigned to.                                  |
+| `Record<T>` | Model with string keys where all the values have type `T` (similar to `Map<string, T>`)    |
+| `unknown`   | A top type in TypeSpec that all types can be assigned to. Values that can have any type should be assigned this value (similar to `any`) |
```
Comment (mikekistler) [inline docs/language-basics/built-in-types.md:53]: Here too, should we say "similar to `any` in TypeScript"?

**Review by mikekistler — APPROVED** [review]
Looks good but I left a couple comments/suggestions I hope you consider.

File: docs/language-basics/built-in-types.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (markcowl) [reply docs/language-basics/built-in-types.md:53]: I think 'Model' has a whole document in this section, as do 'enum', 'Operation', and the like

**Comment by markcowl** [comment]
@mikekistler Updated per feedback, let me know if this is closer to what you had in mind.

**Pull request merged**